src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuardsNew.java

```diff
@@ -0,0 +1,977 @@
+         *
+         * @param c the colony.
+         * @param l the location.
+         */
+        public View(final ColonyView c, @NotNull final BlockPos l)
+        {
+            super(c, l);
+        }
+
+        /**
+         * Creates a new window for the building.
+         *
+         * @return a BlockOut window.
+         */
+        @NotNull
+        @Override
+        public Window getWindow()
+        {
+            return new WindowHutGuardTower(this);
+        }
+
+        @Override
+        public void deserialize(@NotNull final ByteBuf buf)
+        {
+            super.deserialize(buf);
+            assignManually = buf.readBoolean();
+            retrieveOnLowHealth = buf.readBoolean();
+            patrolManually = buf.readBoolean();
+            task = GuardTask.values()[buf.readInt()];
+            final int jobId = buf.readInt();
+            job = jobId == -1 ? null : GuardJob.values()[jobId];
+
+            final int targetSize = buf.readInt();
+            patrolTargets = new ArrayList<>();
+
+            for (int i = 0; i < targetSize; i++)
+            {
+                patrolTargets.add(BlockPosUtil.readFromByteBuf(buf));
+            }
+
+            final int mobSize = buf.readInt();
+            for (int i = 0; i < mobSize; i++)
+            {
+                final MobEntryView mobEntry = MobEntryView.readFromByteBuf(buf);
+                mobsToAttack.add(mobEntry);
+            }
+
+            guardPos = BlockPosUtil.readFromByteBuf(buf);
+        }
+
+        @NotNull
+        @Override
+        public Skill getPrimarySkill()
+        {
+            if (GuardJob.KNIGHT.equals(job))
+            {
+                return Skill.STRENGTH;
+            }
+            else
+            {
+                return Skill.INTELLIGENCE;
+            }
+        }
+
+        @NotNull
+        @Override
+        public Skill getSecondarySkill()
+        {
+            if (GuardJob.KNIGHT.equals(job))
+            {
+                return Skill.ENDURANCE;
+            }
+            else
+            {
+                return Skill.STRENGTH;
+            }
+        }
+
+        public void setAssignManually(final boolean assignManually)
+        {
+            this.assignManually = assignManually;
+        }
+
+        public boolean isAssignManually()
+        {
+            return assignManually;
+        }
+
+        public void setRetrieveOnLowHealth(final boolean retrieveOnLowHealth)
+        {
+            this.retrieveOnLowHealth = retrieveOnLowHealth;
+        }
+
+        public boolean isRetrieveOnLowHealth()
+        {
+            return retrieveOnLowHealth;
+        }
+
+        public void setPatrolManually(final boolean patrolManually)
+        {
+            this.patrolManually = patrolManually;
+        }
+
+        public void setMobsToAttack(final List<MobEntryView> mobsToAttack)
+        {
+            this.mobsToAttack = mobsToAttack;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Store a copy of "mobsToAttack". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)